app/Supervizor/Budget/BudgetFacade.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Supervizor\Budget;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Nette\Caching\Cache;
+use Supervizor\Invoice\InvoiceItem;
+use Supervizor\Storage\CacheStorage;
+
+class BudgetFacade
+{
+    /** @var BudgetRepository */
+    private $budgetGroupRepository;
+
+    /** @var CacheStorage */
+    private $cacheStorage;
+
+
+
+    /**
+     * BudgetFacade constructor.
+     * @param CacheStorage $cacheStorage
+     * @param BudgetRepository $budgetGroupRepository
+     */
+    public function __construct(CacheStorage $cacheStorage, BudgetRepository $budgetGroupRepository)
+    {
+        $this->cacheStorage = $cacheStorage;
+        $this->budgetGroupRepository = $budgetGroupRepository;
+    }
+
+
+
+    public function getBudgetGroups($budgetGroupId = null)
+    {
+        $key = 'BudgetGroup' . $budgetGroupId;
+        $fallback = function (& $dependencies) use ($budgetGroupId) {
+            $dependencies[Cache::EXPIRE] = CacheStorage::EXPIRATION;
+
+            $finalArray = [];
+            $filter = [];
+            $min = 0;
+            $max = 0;
+            $total = 0;
+            if (!is_null($budgetGroupId)) {
+                $filter['id'] = $budgetGroupId;
+            }
+
+            /** @var ArrayCollection $s */
+            $s = $this->budgetGroupRepository->getBudgetGroupRepository()->findBy($filter);
+
+            /** @var BudgetGroup $u */
+            foreach ($s AS $u) {
+                $group = [];
+                $group['id'] = $u->getSlug();
+                $group['nazev'] = $u->getName();
+                $group['popis'] = $u->getDescription();
+                $group['x'] = $u->getX();
+                $group['y'] = $u->getY();
+                $group['barva'] = $u->getColor();
+                $group['max_uhrazeno_udt'] = (new \DateTime)->modify('+2 months')->getTimestamp(); //!FIXME
```

Comment:
OOPS, ted se divam ze jsem zde zapomel !FIXME, neni to spravne protoze tam je fixni rozsah, spravne by v:

* $group['max_uhrazeno_udt'] mel byt timestamp posledniho uhrazeni faktury v budgetGroup
* $group['min_uhrazeno_udt'] mel byt timestamp prvniho uhrazeni faktury v budgetGroup

tyhle rozsahy se pouzivaji v detailu budgetGroup jako min/max pro filtrovani dle datumu